proposals/004-skips/README.md

```diff
@@ -0,0 +1,179 @@
+
+Owner: jiazh@microsoft.com
+
+Impacted Projects:
+
+- [x] spin-operator
+- [ ] `spin kube` plugin
+- [ ] runtime-class-manager
+- [ ] containerd-shim-spin
+- [ ] Governance
+- [ ] Creates a new project
+
+Created: October 16th, 2024
+
+Updated: October 16th, 2024
+
+## Background
+
+The [Runwasi] project supports running Wasm workloads in the same pod as Linux containers. Currently, the SpinApp Custom Resource Definition (CRD) does not support running [Sidecar Containers]. Sidecars are important in use cases where additional services need to run in the same pod as the Spin application, enabling low-latency communication and resource sharing.
+
+[Sidecar Containers]: (https://kubernetes.io/docs/concepts/workloads/pods/sidecar-containers/)
+[Runwasi]: (https://github.com/containerd/runwasi)
+[SpinApp CRD]: (https://www.spinkube.dev/docs/reference/spin-app/)
+[Spin Operators repository]: (https://github.com/spinkube/spin-operator)
+
+### User Stories
+
+- As an user, I have a proprietary solution packaged as a binary that cannot be ported to Wasm. I would like to run this binary as a Linux container. Also, I would like to run this container in the same pod as the Spin application so that they can communicate with each other locally. In addition, I want the sidecar container to start before the Spin application.
+- As a developer, I would like to run [Dapr] sidecar containers in the same pod as the Spin application so that I can leverage Dapr's capabilities, such as service invocation, state management, pub/sub, etc, in my Spin application.
+- As a developer, I would like to run a [Envoy] proxy sidecar container in the same pod as the Spin application so that I can re-use the service mesh capabilities provided by Envoy, such as traffic management, security, observability, etc, in my Spin application.
+
+[Dapr]: (https://dapr.io/)
+[Envoy]: (https://www.envoyproxy.io/)
+
+### Existing applications
+
+Here are some existing prototypes that I have found that runs Wasm workloads side by side with Linux containers:
+
+- [SpinKube - the first look at WebAssembly/WASI application (SpinApp) on Kubernetes](https://dev.to/thangchung/spinkube-the-first-look-at-webassemblywasi-application-spinapp-on-kubernetes-36jd) and [Coffeeshop repo](https://github.com/thangchung/coffeeshop-on-spinkube/tree/feat/global-azure-2024)
+- [Istio Wasm Demo](https://github.com/keithmattix/istio-wasm-demo)
+
+
+## Proposal
+
+
+The main changes are
+
+1. A new field `InitContainers` in the `spinapp_types.go` that defines a list of sidecar containers to be included in the Deployment.
+```go
+// InitContainers defines the list of sidecar containers to be included in the deployment.
+//
+// These containers will not include the main Spin App. They share the Spin App's
+// environment variables and volumes.
+// +kubebuilder:validation:Optional
+InitContainers []corev1.Container `json:"containers,omitempty"`
```

Comment:
@Mossaka does kubelet handle ensuring init containers are executed first?

Reply:
"Init containers must run to completion before the Pod can be ready; sidecar containers continue running during a Pod's lifetime, and do support some probes."

from https://kubernetes.io/docs/concepts/workloads/pods/init-containers/